whale 4 (#28664)
supabase/supabase: apps/www/_blog/2024-08-15-log-drains.mdx

---
title: Introducing Log Drains
description: Log Drains for exporting product logs is now available under Public Alpha
author: ziinc
image: lw12/day-4/log-drains-og.png
thumb: lw12/day-4/log-drains-thumb.png
launchweek: '12'
categories:
  - launch-week
  - developers
  - platform
tags:
  - launch-week
  - o11y
  - logging
date: '2024-08-15'
toc_depth: 3
---

> Today, Supabase is releasing Log Drains for all Team and Enterprise users in Public Alpha.

Reviewer: ```suggestion
Today, Supabase is releasing Log Drains for all Team and Enterprise users.
```

I think we should put " in Public Alpha" at the end under a "Getting started" section or something. These blog posts live on the internet forever, and "Public Alpha" is a snapshot in time

> With Log Drains, developers can now export logs generated by their Supabase products to external destinations, such as Datadog or custom HTTP endpoints. Logs generated by all Supabase products such as the Database, Storage, Realtime and Auth are supported.

Reviewer: ```suggestion
With Log Drains, developers can export logs generated by their Supabase products to external destinations, such as Datadog or custom HTTP endpoints. All logs generated by Supabase products such as the Database, Storage, Realtime and Auth are supported.
```

Beyond providing a single pane of glass inside your existing logging and monitoring system, Log Drains can be used to build additional alerting and observability pipelines. For example, you can ingest Postgres connection logs into your Security Information and Event Management (SIEM) or Intrusion Detection System (IDS) to create custom alerting rules based on events happening in your database.

> This feature also allows for extended retention periods to meet compliance requirements.
> 
> While we continue to improve logging and alerting within the Supabase platform, Log Drains provide an important escape hatch for these advanced use cases today.

Reviewer: ```suggestion
This feature also allows for extended retention periods to meet compliance requirements and provide an important escape hatch for advanced use cases while we continue to improve logging and alerting within the Supabase platform
```

## Configuring Log Drains

Log drains can be set up in the project settings.

<Img alt="Log drains on-boarding page" src="/images/blog/lw12/day-4/log-drains-onboarding.png" />

The initial supported destinations are:

- Datadog Logs
- HTTP Endpoint

Popular destinations like Datadog are supported out of the box. More detailed setup guides are available within the [log drains documentation](https://supabase.com/docs/guides/platform/log-drains).

<Img alt="Datadog Logs drain" src="/images/blog/lw12/day-4/datadog-example.png" />

> For the providers that are not supported out of the box, the HTTP Endpoint drain can be used to send logs to any destination that supports ingestion via HTTP POST requests. For example, you can send logs to an Edge Function, filter, or restructure the logs, and then dispatch them to an external provider. In the following example, we perform a simple `console.log` of the received JSON payload. Detailed setup guide is available under the [Edge Functions guide](https://supabase.com/docs/guides/functions/log-drains).

Reviewer: ```suggestion
For the providers that are not natively supported yet, the HTTP Endpoint drain can be used to send logs to any destination that supports ingestion via HTTP POST requests. For example, you can send logs to an Edge Function, filter, or restructure the logs, and then dispatch them to an external provider. In the following example, we perform a simple `console.log` of the received JSON payload. Detailed setup guide is available under the [Edge Functions guide](https://supabase.com/docs/guides/functions/log-drains).
```

<Img
  alt="HTTP Endpoint drain to Supabase Edge Functions"
  src="/images/blog/lw12/day-4/edge-function-example.png"
/>

> When self-hosting or during local development, Log Drains will be available through the Studio and can be under Project Settings > Log Drains.

Reviewer: ```suggestion
Log Drains are available for self-hosting and local development through the Studio under Project Settings > Log Drains.
```

Reviewer 2: PR not yet released as its blocked

## The Supabase Analytics server

> Log Drains is built on [Logflare](https://github.com/Logflare/logflare), the analytics and observability server of the Supabase stack.

Reviewer: ```suggestion
Log Drains are built into [Logflare](https://github.com/Logflare/logflare), the analytics and observability server of the Supabase stack.
```
